javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java

```diff
@@ -78,14 +79,19 @@ public SourcePrinter println() {
         indented = false;
         return this;
     }
-    
+
     private StringBuilder bufAppend(final String arg) {
         updateCursor(arg);
         return buf.append(arg);
     }
 
+    private static final Pattern SPLIT_NEWLINE_PATTERN;
+    static {
+        SPLIT_NEWLINE_PATTERN = Pattern.compile("\r\n|\r|\n");
```

Comment:
Won't a field initializer do instead of a static block?

Reply:
A static block does only initialize the field if this classes is loaded. If the user does not use it, it won't be loaded at all.

Reply:
How does the performance gain compare to the loss of readability?

Reply:
Your right, the static block does do nothing in this case as this is not a static class and the pattern is always used.